File: TelemView.API/Models/DataForHome.cs
using System.Collections.Generic;

namespace TelemView.API.Models
{
    public class DataForHome
    {
        public ICollection<Course> Courses { get; set; }
        public ICollection<Lecturer> Lecturers { get; set; }
        public ICollection<Organization> Organizations { get; set; }
        public ICollection<OrganizationType> OrganizationTypes { get; set; }
        public ICollection<Student> Students { get; set; }
        public ICollection<Tag> Tags { get; set; }
        public ICollection<Task> Tasks { get; set; }
        public ICollection<ProductType> ProductTypes { get; set; } 
        public ICollection<Year> Years { get; set; } 
        public ICollection<Degree> Degree { get; set; } 
    }
}

File: TelemView-SPA/src/app/_models/dataForHome.ts
import { Lecturer } from './lecturer';
import { Student } from './Student';
import { Tag } from './Tag';
import { Course } from './Course';
import { Organization } from './organization';
import { OrganizationType } from './organizationType';
import { Task } from './task';
import { ProductType } from './productType';
import { Year } from './year';
import { Degree } from './Degree';

//general data for home page filters

export interface DataForHome {
    courses: Course[];
    lecturers: Lecturer[];
    organizations: Organization[];
    organizationTypes: OrganizationType[];
    tasks: Task[];
    productTypes: ProductType[];
    years: Year[];
    degree: Degree[];
}
